src/pages/terraform/downloads/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { ReactElement } from 'react'
+import { GetStaticProps } from 'next'
+import semverGte from 'semver/functions/gte'
+import semverMajor from 'semver/functions/major'
+import semverMinor from 'semver/functions/minor'
+import semverPatch from 'semver/functions/patch'
+import terraformData from 'data/terraform.json'
+import installData from 'data/terraform-install.json'
+import { Product } from 'types/products'
+import { generateStaticProps, GeneratedProps } from 'lib/fetch-release-data'
+import EmptyLayout from 'layouts/empty'
+import ProductDownloadsView from 'views/product-downloads-view'
+import PlaceholderDownloadsView from 'views/placeholder-product-downloads-view'
+
+const VERSION_DOWNLOAD_CUTOFF = '1.0.11'
+
+const TerraformDownloadsPage = (props: GeneratedProps): ReactElement => {
+  if (__config.flags.enable_new_downloads_view) {
+    const { latestVersion, releases } = props
+    return (
+      <ProductDownloadsView
+        latestVersion={latestVersion}
+        pageContent={installData}
+        releases={releases}
+      />
+    )
+  } else {
+    return <PlaceholderDownloadsView />
+  }
+}
+
+/**
+ * Pulled from terraform-website/pages/downloads/index.jsx on 3/9/2022:
+ * https://github.com/hashicorp/terraform-website/blob/master/pages/downloads/index.jsx#L55-L98
+ */
+function filterOldVersions(props) {
+  if (!props?.props?.releases?.versions) return props
+
+  const versions = props.props.releases.versions
+
+  // versions is in the form of { [version]: { ...metadata } }
+  // Filter by arbitrary & reasonable version cutoff
+  const filteredVersions = Object.keys(versions).filter((version) => {
+    if (!semverGte(version, VERSION_DOWNLOAD_CUTOFF)) return false
+    return true
+  })
+
+  /** @type {{[x: string]:{ [y: string]: any}}} */
+  const tree = {}
+
+  /**
+   * Computes the latest patch versions for each major/minor
+   * e.g. given [1.1.2, 1.1.1, 1.1.0, 1.0.9, 1.0.8] -> return [1.1.2, 1.0.9]
+   */
+  filteredVersions.forEach((v) => {
+    const x = semverMajor(v)
+    const y = semverMinor(v)
+    const z = semverPatch(v)
+
+    if (!tree[x]) {
+      tree[x] = { [y]: z }
+    } else if (!tree[x][y]) {
+      tree[x][y] = z
+    } else {
+      tree[x][y] = Math.max(tree[x][y], z)
+    }
+  })
+
+  const newVersions = {}
+
+  Object.entries(tree).forEach(([x, xObj]) => {
+    Object.entries(xObj).forEach(([y, z]) => {
+      const version = `${x}.${y}.${z}`
+      newVersions[version] = versions[version]
+    })
+  })
+
+  props.props.releases.versions = newVersions
+}
+
+export const getStaticProps: GetStaticProps = async () => {
+  const product = terraformData as Product
+  const generatedProps = await generateStaticProps(product)
+
+  filterOldVersions(generatedProps)
```

Comment:
Non-blocking nit, and almost certainly outside the scope of this PR, but the way `filterOldVersions` takes in the whole `{ props }` object and modifies it with a kind of side-effect-y approach, also using the global `VERSION_DOWNLOAD_CUTOFF`, feels a little opaque to me... It feels hard to tell exactly what's happening to generatedProps, and how much of it is affected, without reading the function source of `filterOldVersions`.

A kind of stop-gap-y way to mitigate this might be:

```ts
const rawVersions = generatedProps.props?.releases?.versions
if (rawVersions) {
  const newVersions = filterOldVersions(rawVersions, VERSION_DOWNLOAD_CUTOFF)
  generatedProps.props.releases.versions = newVersions
}

// & also.. modify `filterOldVersions` so it's something like:
// (rawVersions: ???, versionCutoff: string): ??? return `newVersions` 
// ... sorry for the `???`, just not sure on the actual type for incoming versions and don't wanna get it wrong!
```

And maybe another solution, potentially a little clearer, could be to have `fetch-release-data`'s `generateStaticProps` accept a second, optional `options` arg:

```ts
const generatedProps = await generateStaticProps(product, { versionCutoff: VERSION_DOWNLOAD_CUTOFF })
// use something like `filterVersions(rawVersions, versionCutoff)` within `generateStaticProps`
```

Again, def not blocking at all, just wanted to get the idea out there and see if it resonated 👍

Reply:
I had the same thoughts about it being side-effect-y and it definitely took me a minute to figure out what was going on! Updating `filterOldVersions` in a way like you mention where it returns a new array would make it clearer sooner what's happening. I think this had been an urgent request so there wasn't much time for refinement.

I'm less sure about extending `generateStaticProps` since I don't think there are any other products that do this. I think we would want to have two new options:
- `versionCutoff`, like you've proposed
- a boolean variable, maybe `onlyLatestMinorVersions`, that signals to exclude additional versions like what was requested when `filterOldVersions` was created

I've created an Asana ticket for the first part of this, refactoring `filterOldVersions`!
https://app.asana.com/0/1201010428539925/1201944491379825/f